File: src/main/java/com/rsupport/download/Downloadable.java
package com.rsupport.download;

import com.rsupport.download.google.drive.GoogleDriveDownload;

import java.io.File;

public interface Downloadable {
    File execute();

    static void from(String credentialFile, String fileId) {
        from(credentialFile, fileId, null);
    }

    static Downloadable from(String credentialFile, String fileId, String fileName) {
        return new GoogleDriveDownload(credentialFile, fileId, fileName);
    }
}

File: src/main/java/com/rsupport/download/Downloadable.kt
package com.rsupport.download

import com.google.api.client.auth.oauth2.Credential
import com.rsupport.google.drive.GoogleDriveDownload
import java.io.File

interface Downloadable {
    fun execute(): File?

    companion object {
        @JvmStatic
        fun from(credential: Credential, fileId: String) =
                from(credential, fileId, "")

        fun from(credentials : Credential, fileId: String, filePath: String): Downloadable {
            return GoogleDriveDownload(credentials, fileId, filePath)
        }
    }
}